sdk/tests/sdk/whirlpools/whirlpool-impl#closePosition.test.ts

```diff
@@ -0,0 +1,552 @@
+    });
+
+    await tx.buildAndExecute();
+  }
+
+  async function testClosePosition(fixture: WhirlpoolTestFixture) {
+    const { positions, poolInitInfo } = fixture.getInfos();
+
+    const pool = await testCtx.whirlpoolClient.getPool(poolInitInfo.whirlpoolPda.publicKey);
+    const position = await testCtx.whirlpoolClient.getPosition(positions[0].publicKey);
+
+    const positionDataBefore = await testCtx.whirlpoolCtx.fetcher.getPosition(
+      position.getAddress(),
+      true
+    );
+
+    assert.notEqual(positionDataBefore, null);
+
+    const txs = await pool.closePosition(position.getAddress(), Percentage.fromFraction(10, 100));
+
+    for (const tx of txs) {
+      await tx.buildAndExecute();
+    }
+
+    const positionDataAfter = await testCtx.whirlpoolCtx.fetcher.getPosition(
+      position.getAddress(),
+      true
+    );
+
+    assert.equal(positionDataAfter, null);
+  }
+
+  context("when the whirlpool is SPL-only", () => {
+    it("should close a position with no liquidity, fees, or rewards", async () => {
+      const fixture = await new WhirlpoolTestFixture(testCtx.whirlpoolCtx).init({
+        tickSpacing,
+        positions: [
+          { tickLowerIndex, tickUpperIndex, liquidityAmount }, // In range position
+        ],
+      });
+
+      await removeLiquidity(fixture);
+      await testClosePosition(fixture);
+    });
+
+    it("should close a position with only liquidity", async () => {
+      const fixture = await new WhirlpoolTestFixture(testCtx.whirlpoolCtx).init({
+        tickSpacing,
+        positions: [
+          { tickLowerIndex, tickUpperIndex, liquidityAmount }, // In range position
+        ],
+      });
+
+      await testClosePosition(fixture);
+    });
+
+    it("should close a position with only fees", async () => {
+      const fixture = await new WhirlpoolTestFixture(testCtx.whirlpoolCtx).init({
+        tickSpacing,
+        positions: [
+          { tickLowerIndex, tickUpperIndex, liquidityAmount }, // In range position
+        ],
+      });
+
+      await accrueFees(fixture);
+      await removeLiquidity(fixture);
+
+      await testClosePosition(fixture);
+    });
+
+    it("should close a position with only rewards", async () => {
+      const fixture = await new WhirlpoolTestFixture(testCtx.whirlpoolCtx).init({
+        tickSpacing,
+        positions: [
+          { tickLowerIndex, tickUpperIndex, liquidityAmount }, // In range position
+        ],
+        rewards: [
+          {
+            emissionsPerSecondX64: MathUtil.toX64(new Decimal(10)),
+            vaultAmount: new u64(vaultStartBalance),
+          },
+          {
+            emissionsPerSecondX64: MathUtil.toX64(new Decimal(10)),
+            vaultAmount: new u64(vaultStartBalance),
+          },
+          {
+            emissionsPerSecondX64: MathUtil.toX64(new Decimal(10)),
+            vaultAmount: new u64(vaultStartBalance),
+          },
+        ],
+      });
+
+      await accrueRewards(fixture);
```

Comment:
this is true for other "rewards only" tests - no rewards is accrued because a swap must happen for the rewards to accumulate.

To test rewards only, call accrue_fees, then collect_fees, then perform your test on testClosePosition